### ISSUE TITLE
Exception notifications should be disabled by default.

### ISSUE THREAD

**Issue by NSkelsey** [globaleaks/globaleaks-whistleblowing-software#1611]
It was surprising for me to discover that client side exceptions are emailed to `globaleaks-stackexception@lists.globaleaks.org` by default once you configure a connection to an SMTP server.

This should definitely be an opt-in feature, where the administrator chooses to share information to help improve the platform. It **should not** be something that you have to notice and opt-out of. 

All good open source software that wants to collect anonymized information, prompt their users before any kind of information is reported. GL should be no different. I would not be surprised if adopters do not notice that they are sending information about their users to a private mailing list.

**Comment by NSkelsey** [comment]
The UI improvements could either move the notification tab over, or bring it into the 'main' configuration.
As the main configuration stands at the moment, it appears to the unexperienced admin (e.g. Me) that the SMTP email will be receiving emails from the platform not sending them.

The picture below suggest either moving the fields from notif settings into the main ui or visually emphasizing the need to progress through all 4 panes to properly setup email notifications.

<img width="798" alt="screen shot 2016-03-25 at 5 19 13 pm" src="https://cloud.githubusercontent.com/assets/1095794/14054239/0cf1bc3a-f2ae-11e5-90a9-3a5bd6f9fec4.png">

**Comment by evilaliv3** [comment]
I agree!

I would rather instead of disabling it:
1. change the aim of the feature and asking if the user want to enable it during the wizard in an opt-int like way like a lot of softwares do. It should be asked if the user want to participate in our secure and anonymous collection of software exeception that would allow us to be proactive in fixing unexpected faults in the application.
2. instead of an SMTP delivery i would go for implementing a simple HTTPS POST on a sever were to handle them; in past i added the possibility that notifications could be encrypted; this is intendend for the admin to keep track by himself of the exceptions in a secoure menner but i would not find well to hadcode in the software a PGP key as it will be tricky to renew it as it will expire.

what do you think @NSkelsey ?

\cc @fpietrosanti

**Comment by fpietrosanti** [comment]
Because having exception notification handling is a very powerful tool to knows when some piece of the software doesn't work well in a preemptive way, i'd suggest to gives the option to enable it (with proper informative messages, like Firefox o various App ask you about crash handling) at the Activation Wizard stage.

**Comment by evilaliv3** [comment]
ok so, repeating what above with not additional info i think to understand an "i agree".

what is your vision about HTTPS POST or PGP email? :P

**Comment by fpietrosanti** [comment]
HTTPS POST are useful to catch Notification's related exceptions, that otherwise would not be available, but email gives the flexibility also to the sysadmin to receive email notification of exceptions.

Probably both could be useful, like sending to GlobaLeaks via HTTPS-post and to the sysadmin that want to know if/when something doesn't work, but only have email available. 

An HTTPS POST, could be a useful also for some wider handling of Exception Handling and Error Reporting to use #437 or https://github.com/errbit/errbit (that's AirBrake compatibile) or the Cloud AirBrake https://help.airbrake.io/kb/integrations/github but that require additional work

**Comment by evilaliv3** [comment]
i agree; we could have both HTTP POST (for us) and eventually also SMTP for the admin.

i instead against to the usage of cloud services for storing exceptions: e.g. the exception leaks the user agent, the language, the timing and possibilly other information and the failure can happen after the whistleblower already blown the whistle so that that traces could matter.

**Comment by NSkelsey** [comment]
@evilaliv3 I agree with you that there is no reason to integrate a cloud service. I do not like the idea of integrating an external API like airbrake. I think that the safest method would be for the GL server to generate the POST to a Hermes server, through a TOR circuit.

I think that @fpietrosanti's proposal in #437 is the right way to approach this in general. Let the user decide how important it is for the Hermes team to know about an exception. This lets them make their own decisions about anonymity versus supporting the platform.

**Comment by evilaliv3** [comment]
Closing since exception notifications were disabled by default long time ago.
